### ISSUE TITLE
Unexpected cursor position after selecting a word by double clicking it

### ISSUE THREAD

**Issue by sh2ka** [zim-desktop-wiki/zim-desktop-wiki#2116]
**Describe the bug**
It is expected to have a cursor position at the end of selection created by double clicking a word. But ZIM doesn't work that way and the cursor is set in front of the selection that is not expected at all.

**To Reproduce**
1. double click the word in ZIM editor
2. press Shift+Right 
3. you will see that selection decreased at the left instead of increasing at the right

**Expected behavior**
A character at the right of the selection will selected as well so the selection will increase at the right. 

**Screenshots**
![example](https://user-images.githubusercontent.com/2161306/182025698-3cb0480f-940a-444e-a018-b9a61111e5cb.gif)
Gif file demonstrates how Shift+Left/Shift+Right works for double click selection and for simple mouse pointer selection.

**Environment (please complete the following information):**
 - Version 0.74.3 (FlatPack)
 - OS: linux mint cinnamon 20.3
 - Language: en_US.UTF-8

**Comment by introt** [comment]
This is the toolkit's default behavior, and can be observed in other apps using it (eg. `gedit`).

Looking through the API, there doesn't seem to be an obvious way to change this without unsustainable hacks.

As a workaround I offer a workflow that avoids this issue altogether, and should work in both directions everywhere:

1. Single-click with the mouse to move the cursor into a word
2. Press `Ctrl` + `Left` to move the cursor to the beginning of the word
3. Start holding down `Shift`
4. Press `Ctrl` + `Right` to move the cursor to the end of the word
5. Use the arrow keys to adjust the selection to your liking
6. Release `Shift`

Swap `Left` and `Right` to do this in the other direction.

**Comment by sh2ka** [comment]
Ok, I've reproduced that in gedit, but still it is not expected behavior as of my experience as a user. Even javascript editor on this page works as expected. I found an article https://stackoverflow.com/questions/66716616/set-cursor-at-end-of-gtk-entry that could be related to the problem. A solution provided there is to use function gtk_editable_set_position(editable, -1). So you could override double click mouse event and move cursor to the end of the selection to make the editor a little bit more comfortable.

**Comment by introt** [comment]
I agree with you, but sans ugly hacks I haven't yet found a way to fix this downstream.

What you linked is not related to the problem - it solves a related problem. The function moves the cursor to the end of the whole editable - which would in our application mean the end of the page - not the end of the selection, and doesn't delve into hooking up to any mouse events.

However, it's certainly possible that there's a sane solution that I've overlooked. For example, if we used one, the [Gtk.Editable](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Editable.html)'s [get_selection_bounds](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Editable.html#Gtk.Editable.get_selection_bounds) returns negative numbers for backwards selections, so it'd be trivial to swap those around after a double click event (which should also not trigger when double-clicking something else while having text selected etc. - that's the part more likely to introduce new issues).

If you feel like giving the API a second look with fresh eyes, here's links to [Gtk.TextView](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/TextView.html) and [Gtk.TextBuffer](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/TextBuffer.html#Gtk.TextBuffer.get_selection_bounds) (whose `get_selection_bounds` didn't describe the aforementioned behavior...) to get you started. Hit me up with your findings and I'll see if we can rig something up!
